// stream-chat-react-native-core/lib/commonjs/store/schema.js
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tables = void 0;
var tables = {
  channelQueries: {
    columns: {
      cids: 'TEXT',
      id: 'TEXT'
    },
    primaryKey: ['id']
  },
  channels: {
    columns: {
      autoTranslationEnabled: 'BOOLEAN',
      autoTranslationLanguage: 'TEXT',
      cid: 'TEXT',
      config: 'TEXT',
      cooldown: 'BOOLEAN',
      createdAt: 'TEXT',
      createdById: 'TEXT',
      deletedAt: 'TEXT',
      disabled: 'BOOLEAN DEFAULT FALSE',
      extraData: 'TEXT',
      frozen: 'BOOLEAN',
      hidden: 'BOOLEAN',
      id: 'TEXT',
      invites: 'TEXT',
      lastMessageAt: 'TEXT',
      memberCount: 'INTEGER',
      muted: 'BOOLEAN DEFAULT FALSE',
      ownCapabilities: 'TEXT',
      team: 'TEXT',
      truncatedAt: 'TEXT',
      truncatedBy: 'TEXT',
      truncatedById: 'TEXT',
      type: 'TEXT',
      updatedAt: 'TEXT'
    },
    primaryKey: ['cid']
  },
  members: {
    columns: {
      banned: 'BOOLEAN DEFAULT FALSE',
      channelRole: 'TEXT',
      cid: 'TEXT NOT NULL',
      createdAt: 'TEXT',
      inviteAcceptedAt: 'TEXT',
      invited: 'BOOLEAN',
      inviteRejectedAt: 'TEXT',
      isModerator: 'BOOLEAN',
      role: 'TEXT',
      shadowBanned: 'BOOLEAN DEFAULT FALSE',
      updatedAt: 'TEXT',
      userId: 'TEXT'
    },
    foreignKeys: [{
      column: 'cid',
      onDeleteAction: 'CASCADE',
      referenceTable: 'channels',
      referenceTableColumn: 'cid'
    }],
    indexes: [{
      columns: ['cid', 'userId'],
      name: 'index_members',
      unique: false
    }],
    primaryKey: ['cid', 'userId']
  },
  messages: {
    columns: {
      attachments: 'TEXT',
      cid: 'TEXT NOT NULL',
      createdAt: 'TEXT',
      deletedAt: 'TEXT',
      extraData: 'TEXT',
      id: 'TEXT',
      messageTextUpdatedAt: 'TEXT',
      poll_id: 'TEXT',
      reactionGroups: 'TEXT',
      text: "TEXT DEFAULT ''",
      type: 'TEXT',
      updatedAt: 'TEXT',
      userId: 'TEXT'
    },
    foreignKeys: [{
      column: 'cid',
      onDeleteAction: 'CASCADE',
      referenceTable: 'channels',
      referenceTableColumn: 'cid'
    }],
    indexes: [{
      columns: ['cid', 'userId'],
      name: 'index_messages',
      unique: false
    }],
    primaryKey: ['id']
  },
  pendingTasks: {
    columns: {
      channelId: 'TEXT',
      channelType: 'TEXT',
      createdAt: 'TEXT',
      id: 'INTEGER PRIMARY KEY AUTOINCREMENT',
      messageId: 'TEXT',
      payload: 'TEXT',
      type: 'TEXT'
    }
  },
  poll: {
    columns: {
      allow_answers: 'BOOLEAN DEFAULT FALSE',
      allow_user_suggested_options: 'BOOLEAN DEFAULT FALSE',
      answers_count: 'INTEGER DEFAULT 0',
      created_at: 'TEXT',
      created_by: 'TEXT',
      created_by_id: 'TEXT',
      description: 'TEXT',
      enforce_unique_vote: 'BOOLEAN DEFAULT FALSE',
      id: 'TEXT NOT NULL',
      is_closed: 'BOOLEAN DEFAULT FALSE',
      latest_answers: 'TEXT',
      latest_votes_by_option: 'TEXT',
      max_votes_allowed: 'INTEGER DEFAULT 1',
      name: 'TEXT',
      options: 'TEXT',
      own_votes: 'TEXT',
      updated_at: 'TEXT',
      vote_count: 'INTEGER DEFAULT 0',
      vote_counts_by_option: 'TEXT',
      voting_visibility: 'TEXT'
    },
    primaryKey: ['id']
  },
  reactions: {
    columns: {
      createdAt: 'TEXT',
      extraData: 'TEXT',
      messageId: 'TEXT',
      score: 'INTEGER DEFAULT 0',
      type: 'TEXT',
      updatedAt: 'TEXT',
      userId: 'TEXT'
    },
    foreignKeys: [{
      column: 'messageId',
      onDeleteAction: 'CASCADE',
      referenceTable: 'messages',
      referenceTableColumn: 'id'
    }],
    indexes: [{
      columns: ['messageId', 'userId'],
      name: 'index_reaction',
      unique: false
    }],
    primaryKey: ['messageId', 'userId', 'type']
  },
  reads: {
    columns: {
      cid: 'TEXT NOT NULL',
      lastRead: 'TEXT NOT NULL',
      unreadMessages: 'INTEGER DEFAULT 0',
      userId: 'TEXT'
    },
    indexes: [{
      columns: ['cid', 'userId'],
      name: 'index_reads_cid',
      unique: false
    }],
    primaryKey: ['userId', 'cid']
  },
  users: {
    columns: {
      banned: 'BOOLEAN DEFAULT FALSE',
      createdAt: 'TEXT',
      extraData: 'TEXT',
      id: 'TEXT',
      lastActive: 'TEXT',
      online: 'INTEGER',
      role: 'TEXT',
      updatedAt: 'TEXT'
    },
    indexes: [{
      columns: ['id'],
      name: 'index_users_id',
      unique: true
    }],
    primaryKey: ['id']
  },
  userSyncStatus: {
    columns: {
      appSettings: 'TEXT',
      lastSyncedAt: 'TEXT',
      userId: 'TEXT'
    },
    primaryKey: ['userId']
  }
};
exports.tables = tables;
