// stream-chat-react-native-core/lib/module/components/ChannelPreview/hooks/useLatestMessagePreview.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useLatestMessagePreview = exports.MessageReadStatus = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _react = require("react");
var _ChatContext = require("../../../contexts/chatContext/ChatContext");
var _TranslationContext = require("../../../contexts/translationContext/TranslationContext");
var _hooks = require("../../../hooks");
var _useTranslatedMessage = require("../../../hooks/useTranslatedMessage");
var _utils = require("../../../utils/utils");
var selector = function selector(nextValue) {
  return {
    createdBy: nextValue.created_by,
    latestVotesByOption: nextValue.latest_votes_by_option,
    name: nextValue.name
  };
};
var getMessageSenderName = function getMessageSenderName(message, currentUserId, t, membersLength) {
  var _message$user;
  if ((message == null ? void 0 : (_message$user = message.user) == null ? void 0 : _message$user.id) === currentUserId) {
    return t('You');
  }
  if (membersLength > 2) {
    var _message$user2, _message$user3, _message$user4;
    return (message == null ? void 0 : (_message$user2 = message.user) == null ? void 0 : _message$user2.name) || (message == null ? void 0 : (_message$user3 = message.user) == null ? void 0 : _message$user3.username) || (message == null ? void 0 : (_message$user4 = message.user) == null ? void 0 : _message$user4.id) || '';
  }
  return '';
};
var getMentionUsers = function getMentionUsers(mentionedUser) {
  if (Array.isArray(mentionedUser)) {
    var mentionUserString = mentionedUser.reduce(function (acc, cur) {
      var userName = cur.name || cur.id || '';
      if (userName) {
        acc += "".concat(acc.length ? '|' : '', "@").concat(userName);
      }
      return acc;
    }, '');
    return mentionUserString.replace(/[.*+?^${}()|[\]\\]/g, function (match) {
      return '\\' + match;
    });
  }
  return '';
};
var getLatestMessageDisplayText = function getLatestMessageDisplayText(channel, client, message, t, pollState) {
  var _message$attachments;
  if (!message) return [{
    bold: false,
    text: t('Nothing yet...')
  }];
  var isMessageTypeDeleted = message.type === 'deleted';
  if (isMessageTypeDeleted) return [{
    bold: false,
    text: t('Message deleted')
  }];
  var currentUserId = client == null ? void 0 : client.userID;
  var members = Object.keys(channel.state.members);
  var messageSender = getMessageSenderName(message, currentUserId, t, members.length);
  var messageSenderText = messageSender ? "".concat(messageSender === t('You') ? '' : '@').concat(messageSender, ": ") : '';
  var boldOwner = messageSenderText.includes('@');
  if (message.text) {
    var shortenedText = message.text.substring(0, 100).replace(/\n/g, ' ');
    var mentionedUsers = getMentionUsers(message.mentioned_users);
    var regEx = new RegExp("^(".concat(mentionedUsers, ")"));
    return [{
      bold: boldOwner,
      text: messageSenderText
    }].concat((0, _toConsumableArray2["default"])(shortenedText.split('').reduce(function (acc, cur, index) {
      if (cur === '@' && mentionedUsers && regEx.test(shortenedText.substring(index))) {
        acc.push({
          bold: true,
          text: cur
        });
      } else if (mentionedUsers && regEx.test(acc[acc.length - 1].text)) {
        acc.push({
          bold: false,
          text: cur
        });
      } else {
        acc[acc.length - 1].text += cur;
      }
      return acc;
    }, [{
      bold: false,
      text: ''
    }])));
  }
  if (message.command) {
    return [{
      bold: boldOwner,
      text: messageSenderText
    }, {
      bold: false,
      text: "/".concat(message.command)
    }];
  }
  if ((_message$attachments = message.attachments) != null && _message$attachments.length) {
    return [{
      bold: boldOwner,
      text: messageSenderText
    }, {
      bold: false,
      text: t('🏙 Attachment...')
    }];
  }
  if (message.poll_id && pollState) {
    var _latestVotes$, _previewUser, _previewUser2;
    var createdBy = pollState.createdBy,
      latestVotesByOption = pollState.latestVotesByOption,
      name = pollState.name;
    var latestVotes;
    if (latestVotesByOption) {
      latestVotes = Object.values(latestVotesByOption).map(function (votes) {
        return votes == null ? void 0 : votes[0];
      }).sort(function (a, b) {
        return new Date(b.created_at).getTime() - new Date(a.created_at).getTime();
      });
    }
    var previewAction = 'created';
    var previewUser = createdBy;
    if (latestVotes && latestVotes.length && (_latestVotes$ = latestVotes[0]) != null && _latestVotes$.user) {
      var _latestVotes$2;
      previewAction = 'voted';
      previewUser = (_latestVotes$2 = latestVotes[0]) == null ? void 0 : _latestVotes$2.user;
    }
    var previewMessage = "".concat(client.userID === ((_previewUser = previewUser) == null ? void 0 : _previewUser.id) ? 'You' : (_previewUser2 = previewUser) == null ? void 0 : _previewUser2.name, " ").concat(previewAction, ": ").concat(name);
    return [{
      bold: false,
      text: '📊 '
    }, {
      bold: false,
      text: previewMessage
    }];
  }
  return [{
    bold: boldOwner,
    text: messageSenderText
  }, {
    bold: false,
    text: t('Empty message...')
  }];
};
var MessageReadStatus = function (MessageReadStatus) {
  MessageReadStatus[MessageReadStatus["NOT_SENT_BY_CURRENT_USER"] = 0] = "NOT_SENT_BY_CURRENT_USER";
  MessageReadStatus[MessageReadStatus["UNREAD"] = 1] = "UNREAD";
  MessageReadStatus[MessageReadStatus["READ"] = 2] = "READ";
  return MessageReadStatus;
}({});
exports.MessageReadStatus = MessageReadStatus;
var getLatestMessageReadStatus = function getLatestMessageReadStatus(channel, client, message, readEvents) {
  var _message$user5;
  var currentUserId = client.userID;
  if (!message || currentUserId !== ((_message$user5 = message.user) == null ? void 0 : _message$user5.id) || readEvents === false) {
    return MessageReadStatus.NOT_SENT_BY_CURRENT_USER;
  }
  var readList = Object.assign({}, channel.state.read);
  if (currentUserId) {
    delete readList[currentUserId];
  }
  var messageUpdatedAt = message.updated_at ? typeof message.updated_at === 'string' ? new Date(message.updated_at) : message.updated_at : undefined;
  return Object.values(readList).some(function (_ref) {
    var last_read = _ref.last_read;
    return messageUpdatedAt && messageUpdatedAt < last_read;
  }) ? MessageReadStatus.READ : MessageReadStatus.UNREAD;
};
var getLatestMessagePreview = function getLatestMessagePreview(params) {
  var channel = params.channel,
    client = params.client,
    lastMessage = params.lastMessage,
    pollState = params.pollState,
    readEvents = params.readEvents,
    t = params.t;
  var messages = channel.state.messages;
  if (!messages.length && !lastMessage) {
    return {
      created_at: '',
      messageObject: undefined,
      previews: [{
        bold: false,
        text: t('Nothing yet...')
      }],
      status: MessageReadStatus.NOT_SENT_BY_CURRENT_USER
    };
  }
  var channelStateLastMessage = messages.length ? messages[messages.length - 1] : undefined;
  var message = lastMessage !== undefined ? lastMessage : channelStateLastMessage;
  return {
    created_at: message == null ? void 0 : message.created_at,
    messageObject: message,
    previews: getLatestMessageDisplayText(channel, client, message, t, pollState),
    status: getLatestMessageReadStatus(channel, client, message, readEvents)
  };
};
var useLatestMessagePreview = function useLatestMessagePreview(channel, forceUpdate, lastMessage) {
  var _lastMessage$poll_id, _useStateStore;
  var _useChatContext = (0, _ChatContext.useChatContext)(),
    client = _useChatContext.client;
  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
    t = _useTranslationContex.t;
  var channelConfigExists = typeof (channel == null ? void 0 : channel.getConfig) === 'function';
  var translatedLastMessage = (0, _useTranslatedMessage.useTranslatedMessage)(lastMessage);
  var channelLastMessageString = translatedLastMessage ? (0, _utils.stringifyMessage)(translatedLastMessage) : '';
  var _useState = (0, _react.useState)(true),
    _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
    readEvents = _useState2[0],
    setReadEvents = _useState2[1];
  var _useState3 = (0, _react.useState)({
      created_at: '',
      messageObject: undefined,
      previews: [{
        bold: false,
        text: ''
      }],
      status: MessageReadStatus.NOT_SENT_BY_CURRENT_USER
    }),
    _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
    latestMessagePreview = _useState4[0],
    setLatestMessagePreview = _useState4[1];
  var readStatus = getLatestMessageReadStatus(channel, client, translatedLastMessage, readEvents);
  (0, _react.useEffect)(function () {
    if (channelConfigExists) {
      var _channel$getConfig;
      var read_events = !channel.disconnected && !!(channel != null && channel.id) && ((_channel$getConfig = channel.getConfig()) == null ? void 0 : _channel$getConfig.read_events);
      if (typeof read_events === 'boolean') {
        setReadEvents(read_events);
      }
    }
  }, [channelConfigExists]);
  var pollId = (_lastMessage$poll_id = lastMessage == null ? void 0 : lastMessage.poll_id) != null ? _lastMessage$poll_id : '';
  var poll = client.polls.fromState(pollId);
  var pollState = (_useStateStore = (0, _hooks.useStateStore)(poll == null ? void 0 : poll.state, selector)) != null ? _useStateStore : {};
  var createdBy = pollState.createdBy,
    latestVotesByOption = pollState.latestVotesByOption,
    name = pollState.name;
  (0, _react.useEffect)(function () {
    return setLatestMessagePreview(getLatestMessagePreview({
      channel: channel,
      client: client,
      lastMessage: translatedLastMessage,
      pollState: pollState,
      readEvents: readEvents,
      t: t
    }));
  }, [channelLastMessageString, forceUpdate, readEvents, readStatus, latestVotesByOption, createdBy, name]);
  return latestMessagePreview;
};
exports.useLatestMessagePreview = useLatestMessagePreview;
