// stream-chat-react-native-core/lib/commonjs/components/MessageInput/MoreOptionsButton.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MoreOptionsButton = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNativeGestureHandler = require("react-native-gesture-handler");
var _ThemeContext = require("../../contexts/themeContext/ThemeContext");
var _CircleRight = require("../../icons/CircleRight");
var _jsxRuntime = require("react/jsx-runtime");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/MessageInput/MoreOptionsButton.tsx";
var MoreOptionsButton = function MoreOptionsButton(props) {
  var handleOnPress = props.handleOnPress;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    accent_blue = _useTheme$theme.colors.accent_blue,
    moreOptionsButton = _useTheme$theme.messageInput.moreOptionsButton;
  return (0, _jsxRuntime.jsx)(_reactNativeGestureHandler.TouchableOpacity, {
    hitSlop: {
      bottom: 15,
      left: 15,
      right: 15,
      top: 15
    },
    onPress: handleOnPress,
    style: [moreOptionsButton],
    testID: "more-options-button",
    children: (0, _jsxRuntime.jsx)(_CircleRight.CircleRight, {
      pathFill: accent_blue
    })
  });
};
exports.MoreOptionsButton = MoreOptionsButton;
MoreOptionsButton.displayName = 'MoreOptionsButton{messageInput}';
