// stream-chat-react-native-core/lib/commonjs/utils/ACITriggerSettings.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isMentionTrigger = exports.isEmojiTrigger = exports.isCommandTrigger = exports.ACITriggerSettings = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _constants = require("./constants");
var _queryMembers = require("./queryMembers");
var _queryUsers = require("./queryUsers");
var getCommands = function getCommands(channel) {
  var _channel$getConfig;
  return ((_channel$getConfig = channel.getConfig()) == null ? void 0 : _channel$getConfig.commands) || [];
};
var isCommandTrigger = function isCommandTrigger(trigger) {
  return trigger === '/';
};
exports.isCommandTrigger = isCommandTrigger;
var isEmojiTrigger = function isEmojiTrigger(trigger) {
  return trigger === ':';
};
exports.isEmojiTrigger = isEmojiTrigger;
var isMentionTrigger = function isMentionTrigger(trigger) {
  return trigger === '@';
};
exports.isMentionTrigger = isMentionTrigger;
var ACITriggerSettings = function ACITriggerSettings(_ref) {
  var channel = _ref.channel,
    client = _ref.client,
    emojiSearchIndex = _ref.emojiSearchIndex,
    onMentionSelectItem = _ref.onMentionSelectItem;
  return {
    '/': {
      dataProvider: function dataProvider(query, text, onReady) {
        var options = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : {};
        try {
          if (text.indexOf('/') !== 0) return [];
          var _options$limit = options.limit,
            limit = _options$limit === void 0 ? _constants.defaultAutoCompleteSuggestionsLimit : _options$limit;
          var selectedCommands = !query ? getCommands(channel) : getCommands(channel).filter(function (command) {
            var _command$name;
            return query && ((_command$name = command.name) == null ? void 0 : _command$name.indexOf(query)) !== -1;
          });
          selectedCommands.sort(function (a, b) {
            var _a$name, _b$name;
            var nameA = ((_a$name = a.name) == null ? void 0 : _a$name.toLowerCase()) || '';
            var nameB = ((_b$name = b.name) == null ? void 0 : _b$name.toLowerCase()) || '';
            if (query && nameA.indexOf(query) === 0) {
              nameA = "0".concat(nameA);
            }
            if (query && nameB.indexOf(query) === 0) {
              nameB = "0".concat(nameB);
            }
            if (nameA < nameB) return -1;
            if (nameA > nameB) return 1;
            return 0;
          });
          var result = selectedCommands.slice(0, limit);
          if (onReady) {
            onReady(result, query);
          }
          return result;
        } catch (error) {
          console.warn('Error querying commands while using "/":', error);
          throw error;
        }
      },
      output: function output(entity) {
        return {
          caretPosition: 'next',
          key: "".concat(entity.name),
          text: "/".concat(entity.name)
        };
      },
      type: 'command'
    },
    ':': {
      dataProvider: function () {
        var _dataProvider = (0, _asyncToGenerator2["default"])(_regenerator["default"].mark(function _callee(query, _, onReady) {
          var _yield$emojiSearchInd, emojis;
          return _regenerator["default"].wrap(function _callee$(_context) {
            while (1) switch (_context.prev = _context.next) {
              case 0:
                _context.prev = 0;
                if (query) {
                  _context.next = 3;
                  break;
                }
                return _context.abrupt("return", []);
              case 3:
                _context.next = 5;
                return emojiSearchIndex == null ? void 0 : emojiSearchIndex.search(query);
              case 5:
                _context.t0 = _yield$emojiSearchInd = _context.sent;
                if (!(_context.t0 != null)) {
                  _context.next = 10;
                  break;
                }
                _context.t1 = _yield$emojiSearchInd;
                _context.next = 11;
                break;
              case 10:
                _context.t1 = [];
              case 11:
                emojis = _context.t1;
                if (onReady) {
                  onReady(emojis, query);
                }
                return _context.abrupt("return", emojis);
              case 16:
                _context.prev = 16;
                _context.t2 = _context["catch"](0);
                console.warn('Error querying emojis while using ":":', _context.t2);
                throw _context.t2;
              case 20:
              case "end":
                return _context.stop();
            }
          }, _callee, null, [[0, 16]]);
        }));
        function dataProvider(_x, _x2, _x3) {
          return _dataProvider.apply(this, arguments);
        }
        return dataProvider;
      }(),
      output: function output(entity) {
        return {
          caretPosition: 'next',
          key: entity.name,
          text: entity.unicode
        };
      },
      type: 'emoji'
    },
    '@': {
      callback: function callback(item) {
        onMentionSelectItem(item);
      },
      dataProvider: function dataProvider(query, _, onReady) {
        var options = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : {
          limit: _constants.defaultAutoCompleteSuggestionsLimit,
          mentionAllAppUsersEnabled: false,
          mentionAllAppUsersQuery: _constants.defaultMentionAllAppUsersQuery
        };
        try {
          if (!query) return [];
          if (options != null && options.mentionAllAppUsersEnabled) {
            return (0, _queryUsers.queryUsersDebounced)(client, query, function (data) {
              if (onReady) {
                onReady(data, query);
              }
            }, {
              limit: options.limit,
              mentionAllAppUsersQuery: options.mentionAllAppUsersQuery
            });
          }
          if (Object.values(channel.state.members).length < 100) {
            var users = (0, _queryMembers.getMembersAndWatchers)(channel);
            var matchingUsers = users.filter(function (user) {
              var _user$name;
              if (!query) return true;
              if (user.id === client.userID) {
                return false;
              }
              if (((_user$name = user.name) == null ? void 0 : _user$name.toLowerCase().indexOf(query.toLowerCase())) !== -1) {
                return true;
              }
              if (user.id.toLowerCase().indexOf(query.toLowerCase()) !== -1) {
                return true;
              }
              return false;
            });
            var _data = matchingUsers.slice(0, options == null ? void 0 : options.limit);
            if (onReady) {
              onReady(_data, query);
            }
            return _data;
          }
          return (0, _queryMembers.queryMembersDebounced)(client, channel, query, function (data) {
            if (onReady) {
              onReady(data, query);
            }
          }, {
            limit: options.limit
          });
        } catch (error) {
          console.warn("Error querying users/members while using '@':", error);
          throw error;
        }
      },
      output: function output(entity) {
        return {
          caretPosition: 'next',
          key: entity.id,
          text: "@".concat(entity.name || entity.id)
        };
      },
      type: 'mention'
    }
  };
};
exports.ACITriggerSettings = ACITriggerSettings;
