// stream-chat-react-native-core/lib/commonjs/utils/queryUsers.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.queryUsersDebounced = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _debounce = _interopRequireDefault(require("lodash/debounce"));
var _constants = require("./constants");
var queryUsers = function () {
  var _ref = (0, _asyncToGenerator2["default"])(_regenerator["default"].mark(function _callee(client, query, onReady) {
    var options,
      _options$limit,
      limit,
      _options$mentionAllAp,
      mentionAllAppUsersQuery,
      filters,
      _yield$client$queryUs,
      _users,
      usersWithoutClientUserId,
      _args = arguments;
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) switch (_context.prev = _context.next) {
        case 0:
          options = _args.length > 3 && _args[3] !== undefined ? _args[3] : {};
          if (query) {
            _context.next = 3;
            break;
          }
          return _context.abrupt("return");
        case 3:
          _context.prev = 3;
          _options$limit = options.limit, limit = _options$limit === void 0 ? _constants.defaultAutoCompleteSuggestionsLimit : _options$limit, _options$mentionAllAp = options.mentionAllAppUsersQuery, mentionAllAppUsersQuery = _options$mentionAllAp === void 0 ? _constants.defaultMentionAllAppUsersQuery : _options$mentionAllAp;
          filters = Object.assign({
            $or: [{
              id: {
                $autocomplete: query
              }
            }, {
              name: {
                $autocomplete: query
              }
            }]
          }, mentionAllAppUsersQuery == null ? void 0 : mentionAllAppUsersQuery.filters);
          _context.next = 8;
          return client.queryUsers(filters, Object.assign({
            id: 1
          }, mentionAllAppUsersQuery == null ? void 0 : mentionAllAppUsersQuery.sort), Object.assign({
            limit: limit
          }, mentionAllAppUsersQuery == null ? void 0 : mentionAllAppUsersQuery.options));
        case 8:
          _yield$client$queryUs = _context.sent;
          _users = _yield$client$queryUs.users;
          usersWithoutClientUserId = _users.filter(function (user) {
            return user.id !== client.userID;
          });
          if (onReady && _users) {
            onReady(usersWithoutClientUserId);
          }
          _context.next = 18;
          break;
        case 14:
          _context.prev = 14;
          _context.t0 = _context["catch"](3);
          console.warn('Error querying users:', _context.t0);
          throw _context.t0;
        case 18:
        case "end":
          return _context.stop();
      }
    }, _callee, null, [[3, 14]]);
  }));
  return function queryUsers(_x, _x2, _x3) {
    return _ref.apply(this, arguments);
  };
}();
var queryUsersDebounced = (0, _debounce["default"])(queryUsers, 200, {
  leading: false,
  trailing: true
});
exports.queryUsersDebounced = queryUsersDebounced;
