// stream-chat-react-native-core/lib/commonjs/components/MessageList/MessageSystem.js
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageSystem = void 0;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _ThemeContext = require("../../contexts/themeContext/ThemeContext");
var _TranslationContext = require("../../contexts/translationContext/TranslationContext");
var _getDateString = require("../../utils/i18n/getDateString");
var _jsxRuntime = require("react/jsx-runtime");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/MessageList/MessageSystem.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var MessageSystem = function MessageSystem(props) {
  var _message$text;
  var message = props.message,
    style = props.style;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$color = _useTheme$theme.colors,
    grey = _useTheme$theme$color.grey,
    grey_whisper = _useTheme$theme$color.grey_whisper,
    _useTheme$theme$messa = _useTheme$theme.messageList.messageSystem,
    container = _useTheme$theme$messa.container,
    dateText = _useTheme$theme$messa.dateText,
    line = _useTheme$theme$messa.line,
    text = _useTheme$theme$messa.text,
    textContainer = _useTheme$theme$messa.textContainer;
  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
    t = _useTranslationContex.t,
    tDateTimeParser = _useTranslationContex.tDateTimeParser;
  var createdAt = message.created_at;
  var formattedDate = (0, _react.useMemo)(function () {
    return (0, _getDateString.getDateString)({
      date: createdAt,
      t: t,
      tDateTimeParser: tDateTimeParser,
      timestampTranslationKey: 'timestamp/MessageSystem'
    });
  }, [createdAt, t, tDateTimeParser]);
  return (0, _jsxRuntime.jsxs)(_reactNative.View, {
    style: [styles.container, style, container],
    testID: "message-system",
    children: [(0, _jsxRuntime.jsx)(_reactNative.View, {
      style: [styles.line, {
        backgroundColor: grey_whisper
      }, line]
    }), (0, _jsxRuntime.jsxs)(_reactNative.View, {
      style: [styles.textContainer, textContainer],
      children: [(0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: [styles.text, {
          color: grey
        }, text],
        children: ((_message$text = message.text) == null ? void 0 : _message$text.toUpperCase()) || ''
      }), formattedDate && (0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: [styles.text, {
          color: grey
        }, dateText],
        children: formattedDate.toString().toUpperCase()
      })]
    }), (0, _jsxRuntime.jsx)(_reactNative.View, {
      style: [styles.line, {
        backgroundColor: grey_whisper
      }, line]
    })]
  });
};
exports.MessageSystem = MessageSystem;
MessageSystem.displayName = 'MessageSystem{messageList{messageSystem}}';
var styles = _reactNative.StyleSheet.create({
  container: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    marginBottom: 10
  },
  line: {
    flex: 1,
    height: 0.5
  },
  text: {
    fontSize: 10,
    fontWeight: 'bold',
    textAlign: 'center'
  },
  textContainer: {
    flex: 3,
    marginTop: 10
  }
});
