// stream-chat-react-native-core/lib/commonjs/components/Poll/components/PollModalHeader.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PollModalHeader = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _contexts = require("../../../contexts");
var _icons = require("../../../icons");
var _jsxRuntime = require("react/jsx-runtime");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/Poll/components/PollModalHeader.tsx";
var PollModalHeader = function PollModalHeader(_ref) {
  var onPress = _ref.onPress,
    title = _ref.title;
  var _useTheme = (0, _contexts.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$color = _useTheme$theme.colors,
    black = _useTheme$theme$color.black,
    text_high_emphasis = _useTheme$theme$color.text_high_emphasis,
    white = _useTheme$theme$color.white,
    _useTheme$theme$poll$ = _useTheme$theme.poll.modalHeader,
    container = _useTheme$theme$poll$.container,
    titleStyle = _useTheme$theme$poll$.title;
  return (0, _jsxRuntime.jsxs)(_reactNative.View, {
    style: [styles.container, {
      backgroundColor: white
    }, container],
    children: [(0, _jsxRuntime.jsx)(_reactNative.TouchableOpacity, {
      onPress: onPress,
      children: (0, _jsxRuntime.jsx)(_icons.Back, {
        height: 24,
        pathFill: text_high_emphasis,
        viewBox: "0 0 24 24",
        width: 24
      })
    }), (0, _jsxRuntime.jsx)(_reactNative.Text, {
      numberOfLines: 1,
      style: [styles.title, {
        color: black
      }, titleStyle],
      children: title
    })]
  });
};
exports.PollModalHeader = PollModalHeader;
var styles = _reactNative.StyleSheet.create({
  container: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingHorizontal: 16,
    paddingVertical: 18
  },
  title: {
    fontSize: 16,
    fontWeight: '500',
    marginLeft: 32
  }
});
