// stream-chat-react-native-core/lib/module/components/ChannelPreview/ChannelPreviewStatus.js
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ChannelPreviewStatus = void 0;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _useLatestMessagePreview = require("./hooks/useLatestMessagePreview");
var _ThemeContext = require("../../contexts/themeContext/ThemeContext");
var _TranslationContext = require("../../contexts/translationContext/TranslationContext");
var _icons = require("../../icons");
var _getDateString = require("../../utils/i18n/getDateString");
var _jsxRuntime = require("react/jsx-runtime");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/ChannelPreview/ChannelPreviewStatus.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var styles = _reactNative.StyleSheet.create({
  date: {
    fontSize: 12,
    marginLeft: 2,
    textAlign: 'right'
  },
  flexRow: {
    flexDirection: 'row'
  }
});
var ChannelPreviewStatus = function ChannelPreviewStatus(props) {
  var _latestMessagePreview;
  var formatLatestMessageDate = props.formatLatestMessageDate,
    latestMessagePreview = props.latestMessagePreview;
  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
    t = _useTranslationContex.t,
    tDateTimeParser = _useTranslationContex.tDateTimeParser;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$chann = _useTheme$theme.channelPreview,
    checkAllIcon = _useTheme$theme$chann.checkAllIcon,
    checkIcon = _useTheme$theme$chann.checkIcon,
    date = _useTheme$theme$chann.date,
    _useTheme$theme$color = _useTheme$theme.colors,
    accent_blue = _useTheme$theme$color.accent_blue,
    grey = _useTheme$theme$color.grey;
  var created_at = (_latestMessagePreview = latestMessagePreview.messageObject) == null ? void 0 : _latestMessagePreview.created_at;
  var latestMessageDate = created_at ? new Date(created_at) : new Date();
  var formattedDate = (0, _react.useMemo)(function () {
    return (0, _getDateString.getDateString)({
      date: created_at,
      t: t,
      tDateTimeParser: tDateTimeParser,
      timestampTranslationKey: 'timestamp/ChannelPreviewStatus'
    });
  }, [created_at, t, tDateTimeParser]);
  var status = latestMessagePreview.status;
  return (0, _jsxRuntime.jsxs)(_reactNative.View, {
    style: styles.flexRow,
    children: [status === _useLatestMessagePreview.MessageReadStatus.READ ? (0, _jsxRuntime.jsx)(_icons.CheckAll, Object.assign({
      pathFill: accent_blue
    }, checkAllIcon)) : status === _useLatestMessagePreview.MessageReadStatus.UNREAD ? (0, _jsxRuntime.jsx)(_icons.Check, Object.assign({
      pathFill: grey
    }, checkIcon)) : null, (0, _jsxRuntime.jsx)(_reactNative.Text, {
      style: [styles.date, {
        color: grey
      }, date],
      children: formatLatestMessageDate && latestMessageDate ? formatLatestMessageDate(latestMessageDate).toString() : formattedDate
    })]
  });
};
exports.ChannelPreviewStatus = ChannelPreviewStatus;
