// stream-chat-react-native-core/lib/commonjs/components/Poll/components/PollResults/PollResultItem.js
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PollVote = exports.PollResultsItem = void 0;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _streamChat = require("stream-chat");
var _contexts = require("../../../../contexts");
var _getDateString = require("../../../../utils/i18n/getDateString");
var _Avatar = require("../../../Avatar/Avatar");
var _usePollState3 = require("../../hooks/usePollState");
var _Button = require("../Button");
var _jsxRuntime = require("react/jsx-runtime");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/Poll/components/PollResults/PollResultItem.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var PollVote = function PollVote(_ref) {
  var _vote$user, _vote$user2;
  var vote = _ref.vote;
  var _useTranslationContex = (0, _contexts.useTranslationContext)(),
    t = _useTranslationContex.t,
    tDateTimeParser = _useTranslationContex.tDateTimeParser;
  var _usePollState = (0, _usePollState3.usePollState)(),
    votingVisibility = _usePollState.votingVisibility;
  var _useTheme = (0, _contexts.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$color = _useTheme$theme.colors,
    black = _useTheme$theme$color.black,
    text_low_emphasis = _useTheme$theme$color.text_low_emphasis,
    _useTheme$theme$poll$ = _useTheme$theme.poll.results.vote,
    container = _useTheme$theme$poll$.container,
    dateText = _useTheme$theme$poll$.dateText,
    userName = _useTheme$theme$poll$.userName;
  var dateString = (0, _react.useMemo)(function () {
    return (0, _getDateString.getDateString)({
      date: vote.created_at,
      t: t,
      tDateTimeParser: tDateTimeParser,
      timestampTranslationKey: 'timestamp/PollVote'
    });
  }, [vote.created_at, t, tDateTimeParser]);
  var isAnonymous = (0, _react.useMemo)(function () {
    return votingVisibility === _streamChat.VotingVisibility.anonymous;
  }, [votingVisibility]);
  return (0, _jsxRuntime.jsxs)(_reactNative.View, {
    style: [styles.voteContainer, container],
    children: [(0, _jsxRuntime.jsxs)(_reactNative.View, {
      style: {
        flexDirection: 'row'
      },
      children: [!isAnonymous && (_vote$user = vote.user) != null && _vote$user.image ? (0, _jsxRuntime.jsx)(_Avatar.Avatar, {
        image: vote.user.image,
        size: 20
      }, vote.id) : null, (0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: [styles.voteUserName, {
          color: black
        }, userName],
        children: isAnonymous ? t('Anonymous') : (_vote$user2 = vote.user) == null ? void 0 : _vote$user2.name
      })]
    }), (0, _jsxRuntime.jsx)(_reactNative.Text, {
      style: [styles.voteDate, {
        color: text_low_emphasis
      }, dateText],
      children: dateString
    })]
  });
};
exports.PollVote = PollVote;
var PollResultsVoteItem = function PollResultsVoteItem(vote) {
  return (0, _jsxRuntime.jsx)(PollVote, {
    vote: vote
  }, "results_vote_".concat(vote.id));
};
var PollResultsItem = function PollResultsItem(_ref2) {
  var _voteCountsByOption$o, _latestVotesByOption$, _latestVotesByOption$2;
  var option = _ref2.option;
  var _useTranslationContex2 = (0, _contexts.useTranslationContext)(),
    t = _useTranslationContex2.t;
  var _usePollState2 = (0, _usePollState3.usePollState)(),
    latestVotesByOption = _usePollState2.latestVotesByOption,
    voteCountsByOption = _usePollState2.voteCountsByOption;
  var _useTheme2 = (0, _contexts.useTheme)(),
    _useTheme2$theme = _useTheme2.theme,
    _useTheme2$theme$colo = _useTheme2$theme.colors,
    bg_user = _useTheme2$theme$colo.bg_user,
    black = _useTheme2$theme$colo.black,
    _useTheme2$theme$poll = _useTheme2$theme.poll.results.item,
    container = _useTheme2$theme$poll.container,
    headerContainer = _useTheme2$theme$poll.headerContainer,
    title = _useTheme2$theme$poll.title,
    voteCount = _useTheme2$theme$poll.voteCount;
  return (0, _jsxRuntime.jsxs)(_reactNative.View, {
    style: [styles.container, {
      backgroundColor: bg_user
    }, container],
    children: [(0, _jsxRuntime.jsxs)(_reactNative.View, {
      style: [styles.headerContainer, headerContainer],
      children: [(0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: [styles.title, {
          color: black
        }, title],
        children: option.text
      }), (0, _jsxRuntime.jsx)(_reactNative.Text, {
        style: [styles.voteCount, {
          color: black
        }, voteCount],
        children: t('{{count}} votes', {
          count: (_voteCountsByOption$o = voteCountsByOption[option.id]) != null ? _voteCountsByOption$o : 0
        })
      })]
    }), (latestVotesByOption == null ? void 0 : (_latestVotesByOption$ = latestVotesByOption[option.id]) == null ? void 0 : _latestVotesByOption$.length) > 0 ? (0, _jsxRuntime.jsx)(_reactNative.View, {
      style: {
        marginTop: 16
      },
      children: ((_latestVotesByOption$2 = latestVotesByOption == null ? void 0 : latestVotesByOption[option.id]) != null ? _latestVotesByOption$2 : []).slice(0, 5).map(PollResultsVoteItem)
    }) : null, (0, _jsxRuntime.jsx)(_Button.ShowAllVotesButton, {
      option: option
    })]
  });
};
exports.PollResultsItem = PollResultsItem;
var styles = _reactNative.StyleSheet.create({
  container: {
    borderRadius: 12,
    marginBottom: 8,
    paddingHorizontal: 16,
    paddingVertical: 12
  },
  headerContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  title: {
    flex: 1,
    fontSize: 16,
    fontWeight: '500'
  },
  voteContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 8,
    paddingVertical: 8
  },
  voteCount: {
    fontSize: 16,
    marginLeft: 16
  },
  voteDate: {
    fontSize: 14
  },
  voteUserName: {
    fontSize: 14,
    marginLeft: 2
  }
});
