// stream-chat-react-native-core/lib/module/components/Message/MessageSimple/MessageTimestamp.js
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageTimestamp = void 0;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _ThemeContext = require("../../../contexts/themeContext/ThemeContext");
var _TranslationContext = require("../../../contexts/translationContext/TranslationContext");
var _getDateString = require("../../../utils/i18n/getDateString");
var _jsxRuntime = require("react/jsx-runtime");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/Message/MessageSimple/MessageTimestamp.tsx";
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var MessageTimestamp = function MessageTimestamp(props) {
  var formattedDate = props.formattedDate,
    propsTDateTimeParser = props.tDateTimeParser,
    timestamp = props.timestamp,
    _props$timestampTrans = props.timestampTranslationKey,
    timestampTranslationKey = _props$timestampTrans === void 0 ? 'timestamp/MessageTimestamp' : _props$timestampTrans;
  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
    t = _useTranslationContex.t,
    contextTDateTimeParser = _useTranslationContex.tDateTimeParser;
  var tDateTimeParser = propsTDateTimeParser || contextTDateTimeParser;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    grey = _useTheme$theme.colors.grey,
    timestampText = _useTheme$theme.messageSimple.content.timestampText;
  var dateString = (0, _react.useMemo)(function () {
    return (0, _getDateString.getDateString)({
      date: timestamp,
      t: t,
      tDateTimeParser: tDateTimeParser,
      timestampTranslationKey: timestampTranslationKey
    });
  }, [timestamp, t, tDateTimeParser, timestampTranslationKey]);
  if (formattedDate) {
    return (0, _jsxRuntime.jsx)(_reactNative.Text, {
      style: [styles.text, {
        color: grey
      }, timestampText],
      children: formattedDate.toString()
    });
  }
  if (!dateString) return null;
  return (0, _jsxRuntime.jsx)(_reactNative.Text, {
    style: [styles.text, {
      color: grey
    }, timestampText],
    children: dateString.toString()
  });
};
exports.MessageTimestamp = MessageTimestamp;
var styles = _reactNative.StyleSheet.create({
  text: {
    fontSize: 12
  }
});
