// stream-chat-react-native-core/lib/module/utils/addReactionToLocalState.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addReactionToLocalState = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _apis = require("../store/apis");
var _insertReaction = require("../store/apis/insertReaction");
var addReactionToLocalState = function addReactionToLocalState(_ref) {
  var channel = _ref.channel,
    enforceUniqueReaction = _ref.enforceUniqueReaction,
    messageId = _ref.messageId,
    reactionType = _ref.reactionType,
    user = _ref.user;
  var message = channel.state.messages.find(function (_ref2) {
    var id = _ref2.id;
    return id === messageId;
  });
  if (!message) return;
  var reaction = {
    created_at: new Date().toISOString(),
    message_id: messageId,
    type: reactionType,
    updated_at: new Date().toISOString(),
    user: user,
    user_id: user == null ? void 0 : user.id
  };
  var hasOwnReaction = message.own_reactions && message.own_reactions.length > 0;
  if (!message.own_reactions) {
    message.own_reactions = [];
  }
  if (!message.latest_reactions) {
    message.latest_reactions = [];
  }
  if (enforceUniqueReaction) {
    var currentReaction = message.own_reactions[0];
    message.own_reactions = [];
    if (!message.latest_reactions) {
      message.latest_reactions = [];
    }
    message.latest_reactions = message.latest_reactions.filter(function (r) {
      return r.user_id !== user.id;
    });
    if (currentReaction && message.reaction_groups && message.reaction_groups[currentReaction.type] && message.reaction_groups[currentReaction.type].count > 0 && message.reaction_groups[currentReaction.type].sum_scores > 0) {
      message.reaction_groups[currentReaction.type].count = message.reaction_groups[currentReaction.type].count - 1;
      message.reaction_groups[currentReaction.type].sum_scores = message.reaction_groups[currentReaction.type].sum_scores - 1;
    }
    if (!message.reaction_groups) {
      message.reaction_groups = (0, _defineProperty2["default"])({}, reactionType, {
        count: 1,
        first_reaction_at: new Date().toISOString(),
        last_reaction_at: new Date().toISOString(),
        sum_scores: 1
      });
    } else {
      if (!message.reaction_groups[reactionType]) {
        message.reaction_groups[reactionType] = {
          count: 1,
          first_reaction_at: new Date().toISOString(),
          last_reaction_at: new Date().toISOString(),
          sum_scores: 1
        };
      } else {
        message.reaction_groups[reactionType] = Object.assign({}, message.reaction_groups[reactionType], {
          count: message.reaction_groups[reactionType].count + 1,
          last_reaction_at: new Date().toISOString(),
          sum_scores: message.reaction_groups[reactionType].sum_scores + 1
        });
      }
    }
  }
  message.own_reactions = [].concat((0, _toConsumableArray2["default"])(message.own_reactions), [reaction]);
  message.latest_reactions = [].concat((0, _toConsumableArray2["default"])(message.latest_reactions), [reaction]);
  if (enforceUniqueReaction && hasOwnReaction) {
    (0, _apis.updateReaction)({
      message: message,
      reaction: reaction
    });
  } else {
    (0, _insertReaction.insertReaction)({
      message: message,
      reaction: reaction
    });
  }
};
exports.addReactionToLocalState = addReactionToLocalState;
